oak_restricted_kernel_sdk_proc_macro/src/lib.rs

```diff
@@ -0,0 +1,53 @@
+use proc_macro::TokenStream;
+use quote::quote;
+use syn::{parse_macro_input, Item, Result};
+
+#[proc_macro_attribute]
+pub fn entrypoint(_attr: TokenStream, input: TokenStream) -> TokenStream {
+    let input_item = parse_macro_input!(input as syn::Item);
+
+    match process_oak_main(input_item) {
+        Ok(output) => output,
+        Err(error) => error.to_compile_error().into(),
+    }
+}
+
+fn process_oak_main(input_item: Item) -> Result<TokenStream> {
+    let input_fn = match input_item {
+        syn::Item::Fn(fn_item) => Ok(fn_item),
+        _ => Err(syn::Error::new(
+            syn::spanned::Spanned::span(&input_item),
+            "the oak_main macro can only be applied to functions",
```

Comment:
This is not the correct name of the macro I think

Reply:
very good catch.